jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/SecurityProviders.java

```diff
@@ -0,0 +1,66 @@
+/*[INCLUDE-IF CRIU_SUPPORT]*/
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.eclipse.openj9.criu;
+
+import java.security.CRIUConfigurator;
+
+import openj9.internal.criu.InternalCRIUSupport;
+
+/**
+ * Handles the security providers.
+ * The CRIUSECProvider is a security provider that is used as follows when CRIU
+ * is enabled. During the checkpoint phase, all security providers are removed
+ * from the system properties (which are read from security.java) and CRIUSEC is
+ * added to the system properties. The pre-checkpoint hook clears the digests,
+ * to ensure that no state is saved during checkpoint that would be restored
+ * during the restore phase. During the resore phase, CRIUSEC is removed from
+ * the provider list and the other security providers are added back to the
+ * system properties. A new provider list is created from the system properties.
+ */
+public class SecurityProviders {
+
```

Comment:
This class should have an explicit constructor (with the appropriate visibility).